products/mlengine/api.yaml

```diff
@@ -73,3 +91,148 @@ objects:
+        imports: name
+        required: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: |
+          The name specified for the version when it was created.
+
+          The version name must be unique within the model it is created in.
+        required: true
+      - !ruby/object:Api::Type::String
+        name: 'description'
+        description: The description specified for the version when it was created.
+        # isDefault is output-only, but all creation-based providers will override this
+        # to implement `setDefaultVersion` API.
+      - !ruby/object:Api::Type::Boolean
+        name: 'isDefault'
+        description:  If true, this version will be used to handle prediction requests that do not specify a version
+        output: true
+      - !ruby/object:Api::Type::String
+        name: 'deploymentUri'
+        description: The Cloud Storage location of the trained model used to create the version
+        required: true
+      - !ruby/object:Api::Type::Time
+        name: 'createTime'
+        description: The time the version was created.
+        output: true
+      - !ruby/object:Api::Type::Time
+        name: 'lastUseTime'
+        description: The time the version was last used for prediction.
+        output: true
+      - !ruby/object:Api::Type::String
+        name: 'runtimeVersion'
+        description: The AI Platform runtime version to use for this deployment
+      - !ruby/object:Api::Type::Enum
+        name: 'machineType'
+        description: The type of machine on which to serve the model. Currently only applies to online prediction service.
+        values:
+          - mls1-c1-m2
+          - mls1-c4-m2
+      - !ruby/object:Api::Type::Enum
+        name: 'state'
+        description: The state of a version
+        values:
+          - UNKNOWN
+          - READY
+          - CREATING
+          - FAILED
+          - DELETING
+          - UPDATING
+        output: true
+      - !ruby/object:Api::Type::String
+        name: 'errorMessage'
+        description: The details of a failure or cancellation
+        output: true
+      - !ruby/object:Api::Type::Array
+        name: 'packageUris'
+        description: Cloud Storage paths (gs://…) of packages for custom prediction routines or scikit-learn pipelines with custom code.
+        output: true
+        item_type: Api::Type::String
+      - !ruby/object:Api::Type::KeyValuePairs
+        name: 'labels'
+        description: One or more labels that you can add, to organize your model versions
+        # no etag since it's input: true
+      - !ruby/object:Api::Type::Enum
+        name: 'framework'
+        description: The machine learning framework AI Platform uses to train this version of the model
+        values:
+          - FRAMEWORK_UNSPECIFIED
+          - TENSORFLOW
+          - SCIKIT_LEARN
+          - XGBOOST
+      - !ruby/object:Api::Type::String
+        name: 'pythonVersion'
```

Comment:
If there's a fixed set of values for this param, should it be an Enum?

Reply:
fixed!